engine.py

```diff
@@ -181,15 +189,19 @@ def init_engine(self):
             # Log the warning.
             self.log_warning(msg)
 
-        # Make sure we are not running Nuke PLE!
+        # Make sure we are not running Nuke PLE or Non-Commercial!
         if nuke.env.get("ple"):
-            self.log_error("The Nuke Engine does not work with the Nuke PLE!")
+            self.log_error("The Nuke Engine does not work with Nuke PLE!")
+            return
+        elif nuke.env.get("nc"):
+            self.log_error("The Nuke Engine does not work with Nuke Non-Commercial!")
             return
 
-        # Now check that there is a location on disk which corresponds to the context.
-        if self.context.project is None:
+        # Now check that we are at least in a project context. Note that plugin mode
+        # does not require this check since it can operate at the site level.
+        if not self.in_plugin_mode and self.context.project is None:
```

Comment:
Not sure I follow. How can an engine in Toolkit Classic run without a project? That doesn't seem feasible at the moment. I can always remove the code as I don't think it was reachable in the first place.